plugins/inventory/efficientip.py

```diff
@@ -0,0 +1,360 @@
+# -*- Mode: Python; python-indent-offset: 4 -*-
+#
+# Time-stamp: <2020-04-13 21:46:58 alex>
+#
+
+"""
+ EfficientIP inventory for ansible
+ SOLIDserver Device Manager connection to gather devices filtered
+"""
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import ipaddress
+# import pprint
+import logging
+
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.plugins.inventory import Constructable, Cacheable
+from ansible.utils.display import Display
+from ansible.errors import AnsibleError
+
+try:
+    from SOLIDserverRest import adv as sdsadv
+    from SOLIDserverRest import SDSError
+except ImportError:
+    raise AnsibleError('The efficientip dynamic inventory'
+                       ' plugin requires SOLIDserverRest')
+
+logging.basicConfig(format='[%(filename)s:%(lineno)d]'
+                    ' %(levelname)s: %(message)s',
+                    level=logging.INFO)
+
+DOCUMENTATION = '''
+    name: efficientip
+    plugin_type: inventory
+    short_description: inventory based on EfficientIP IPAM
+    requirements:
+        - SOLIDserverRest > 2.0.1
+        - python 3
+    extends_documentation_fragment:
+        - inventory_cache
+    description:
+        - Get inventory hosts from EfficientIP IPAM and Device Manager
+        - Uses a YAML configuration file C(efficientip.(yml|yaml)).
+    author:
+        - Alex Chauvin (@efficientip)
+    options:
+        plugin:
+            description: Token that ensures this is a source file for
+                         the plugin.
+            required: True
+            choices: ['efficientip']
+        api:
+            description: information to get connected to the SOLIDserver
+            required: True
+            type: dictionary
+        space:
+            description: name of the space
+            required: False
+            type: string
+            default: Local
+        filter:
+            description: list of filters to reduce the set of devices
+                         returned from Device Manager. Can mix limit,
+                         device_in_subnet, device_of_class,
+                         device_metadata, device_intf_space. All
+                         filter are additiionnal, all conditions must
+                         match
+            required: False
+            type: dictionary
+        group_metadata:
+            description: regroup devices by metadata
+            required: False
+            type: list
+'''
+
+EXAMPLES = '''
+# example with only credentials to get the whole content of Device Manager
+plugin: efficientip
```

Comment:
Do we need FQCN for new inventory plugin names?

Reply:
We do.

Reply:
done